samples/nrf_desktop/src/modules/usb_state.c

```diff
@@ -39,12 +43,42 @@ static int get_report(struct usb_setup_packet *setup, s32_t *len, u8_t **data)
 		if (request_value[1] == 0x03) {
 			/* Request for feature report */
 			if (request_value[0] == REPORT_ID_USER_CONFIG) {
-				static u8_t status;
-
-				status = atomic_get(&forward_status);
-				*len = sizeof(status);
-				*data = (u8_t *)(&status);
-
+				size_t length = *len;
+				u8_t *buffer = *data;
+
+				struct config_channel_frame frame;
+				int pos = config_channel_report_parse(buffer, length, &frame, true);
+				if (pos < 0) {
+					LOG_WRN("Could not parse report");
+					return pos;
+				}
+
+				bool is_fetch = (frame.status == FORWARD_STATUS_FETCH);
```

Comment:
what if we are in the middle of an operation and status from host will change?

Reply:
Host is required not to do that. It will call set and get report in a sequence, next after completion of the previous. Anyway, here we obtain the status one in "get_report" and then run to completion of this operation.

Reply:
but if host do it will be react correctly? will the old operation be terminated ? is there any timeout on our side?

Reply:
?

Reply:
timeout in firmware added